pkg/pvc/pvc_protected_entity.go

```diff
@@ -283,6 +286,74 @@ func (this PVCProtectedEntity) getProtectedEntityForPV(ctx context.Context, pv *
 				return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
 			}
 		}
+	} else if pv.Spec.VsphereVolume != nil {
+		if pv.Spec.AccessModes[0] != core_v1.ReadWriteOnce {
+			return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
+		}
+		if pv.Annotations[MigratedCSIVolumeAnnotation] == VSphereCSIProvisioner {
+			volumePath := pv.Spec.VsphereVolume.VolumePath
+			this.logger.Infof("Trying to get volume id for migrated volume path %s", volumePath)
+			// Migrated volume must be of type ivd
+			pvPEType := this.getComponentPEType()
+			if pvPEType != astrolabe.IvdPEType {
+				return nil, errors.Errorf("Could not support %s as migrated csi volume type", pvPEType)
+			}
+			//assert pvPEType == astrolabe.IvdPEType
+
+			migrationResourcelist, err := this.getMigrationResourcelist(ctx)
+			if err != nil {
+				return nil, errors.Wrapf(err, "Could not list resource for cnsvspherevolumemigrations")
+			}
+			var pvIDstr string
+			for _, item := range migrationResourcelist.Items {
+				name := item.GetName()
+				// Parse volumeid and volumepath field from cnsvspherevolumemigrations
+				migrationVolumeId, exists, err := unstructured.NestedString(item.UnstructuredContent(), "spec", "volumeid")
+				if err != nil {
+					this.logger.Errorf("Error reading cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+				if !exists {
+					this.logger.Errorf("Error reading field spec.volumeid cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+				migrationVolumePath, exists, err := unstructured.NestedString(item.UnstructuredContent(), "spec", "volumepath")
+				if err != nil {
+					this.logger.Errorf("Error reading cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+				if !exists {
+					this.logger.Errorf("Error reading field spec.volumepath cnsvspherevolumemigrations %s: %v", name, err)
+					continue
+				}
+
+				// If migrationVolumePath field matches volumePath from migrated volume
+				if migrationVolumePath == volumePath {
+					pvIDstr = migrationVolumeId
```

Comment:
you can `break` once a matched volume is found, right?

Reply:
good catach